community/cypher/interpreted-runtime/src/test/scala/org/neo4j/cypher/internal/runtime/interpreted/pipes/ExpandIntoPipeTest.scala

```diff
@@ -60,13 +60,31 @@ class ExpandIntoPipeTest extends CypherFunSuite with PipeTestSupport {
 
   test("should return no relationships for types that have not been defined yet") {
     // given
-    when(query.getRelationshipsForIds(any(), any(), any())).thenAnswer(new Answer[Iterator[Relationship]]{
+    when(query.getRelationshipsForIds(any(), any(), any())).thenAnswer(new Answer[Iterator[Relationship]] {
       override def answer(invocationOnMock: InvocationOnMock): Iterator[Relationship] = {
         val arg = invocationOnMock.getArgument[Option[Array[Int]]](2)
         arg match {
           case None => Iterator.empty
           case Some(array) if array.isEmpty => Iterator.empty
-          case _ => Iterator(relationship1, relationship2)
+          case _ =>
+            val nodeId = invocationOnMock.getArgument[Long](0)
+            val dir = invocationOnMock.getArgument[SemanticDirection](1)
+            if (nodeId == 0) {
+              dir match {
+                case SemanticDirection.INCOMING => Iterator.empty
+                case SemanticDirection.OUTGOING =>
+                  Iterator(relationship1, relationship2, relationship3, selfRelationship)
+
+                case SemanticDirection.BOTH => Iterator(relationship1, relationship2, relationship3, selfRelationship)
+              }
+            }
+            else if (nodeId == 2) {
+              dir match {
+                case SemanticDirection.INCOMING => Iterator(relationship1)
+                case SemanticDirection.OUTGOING => Iterator.empty
+                case SemanticDirection.BOTH => Iterator(relationship1)
+              }
+            } else ???
```

Comment:
I think I'm not understanding something here... where is this new mocking used?

Reply:
The test used to always return the same relationships regardless of input node so we got cases where the returned relationship neither started nor ended on the provided node. The test kind of worked since we were swallowing the null value but we no longer do that. So I changed the test to return the correct relationship given the input node and direction. The `???` is just lazy I guess, I could throw an assertion error instead perhaps?